src/app-logic/url-handling.js

```diff
@@ -180,68 +203,86 @@ export function urlStateToUrlObject(urlState: UrlState): UrlObject {
   const pathParts = [...dataSourceDirs, urlState.selectedTab];
   const { selectedThread } = urlState.profileSpecific;
 
-  // Start with the query parameters that are shown regardless of the active tab.
-  const query: QueryShape = {
+  // Start with the query parameters that are shown regardless of the active panel.
+  let baseQuery;
+  if (urlState.showTabOnly === null) {
```

Comment:
The code inside that if looks new, but actually just moved from below. I had to change the order to make it possible for flow to infer the real type. (otherwise it wasn't possible to understand if it was full view or active tab, because all the fields were common.)